p2pclient/streams.go

```diff
@@ -0,0 +1,109 @@
+package p2pclient
+
+import (
+	"errors"
+	"io"
+	"net"
+
+	ggio "github.com/gogo/protobuf/io"
+	pb "github.com/libp2p/go-libp2p-daemon/pb"
+)
+
+// NewStream initializes a new stream on one of the protocols in protos with
+// the specified peer.
+func (c *Client) NewStream(peer []byte, protos []string) (*pb.StreamInfo, io.ReadWriteCloser, error) {
+	control, err := c.newControlConn()
+	if err != nil {
+		return nil, nil, err
+	}
+	r := ggio.NewDelimitedReader(control, MessageSizeMax)
```

Comment:
I wonder if we might have a case of the delimited reader buffer eating chunks of stream input. Tests will hopefully catch if that's the case.